src/goto-programs/interpreter.cpp

```diff
@@ -500,7 +500,8 @@ exprt interpretert::get_value(
     }
     return result;
   }
-  if(use_non_det && (memory[offset].initialized>=0))
+  if(use_non_det &&
+     memory[offset].initialized!=memory_cellt::initializedt::WRITTEN_BEFORE_READ)
```

Comment:
The statement is a != so it returns nondet when READ_BEFORE_WRITTEN which is the correct case. The UNKNOWN case is in fact READ_BEFORE_WRITTEN because we're reading before a write took place (not sure why it was added).